// @contentful/forma-36-react-components/dist/components/EmptyState/EmptyState.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EmptyState = EmptyState;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Typography = require("../Typography/");

var _excluded = ["className", "children", "testId", "customImageElement", "imageProps", "headingProps", "descriptionProps"];
var styles = {
  "EmptyState": "EmptyState__EmptyState___35Xbk",
  "EmptyState_container": "EmptyState__EmptyState_container___34eoG",
  "EmptyState_illustration": "EmptyState__EmptyState_illustration___3KEC9",
  "EmptyState_element": "EmptyState__EmptyState_element___1cYfd",
  "EmptyState_paragraph": "EmptyState__EmptyState_paragraph___KTUQq"
};

function EmptyState(_ref) {
  var className = _ref.className,
      children = _ref.children,
      _ref$testId = _ref.testId,
      testId = _ref$testId === void 0 ? 'cf-ui-empty-state' : _ref$testId,
      customImageElement = _ref.customImageElement,
      imageProps = _ref.imageProps,
      headingProps = _ref.headingProps,
      descriptionProps = _ref.descriptionProps,
      otherProps = (0, _objectWithoutProperties2["default"])(_ref, _excluded);
  var classNames = (0, _classnames["default"])(styles.EmptyState, className);
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({}, otherProps, {
    className: classNames,
    "data-test-id": testId
  }), /*#__PURE__*/_react["default"].createElement("div", {
    className: styles['EmptyState_container']
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: styles['EmptyState_element']
  }, customImageElement ? customImageElement : imageProps && /*#__PURE__*/_react["default"].createElement("img", {
    src: imageProps.url,
    alt: imageProps.description,
    className: (0, _classnames["default"])(imageProps.className, styles['EmptyState_illustration']),
    style: {
      height: imageProps.height,
      width: imageProps.width
    }
  })), /*#__PURE__*/_react["default"].createElement(_Typography.Heading, {
    element: headingProps.elementType ? headingProps.elementType : 'h1',
    className: styles['EmptyState_element']
  }, headingProps.text), /*#__PURE__*/_react["default"].createElement(_Typography.Paragraph, {
    element: descriptionProps.elementType ? descriptionProps.elementType : 'p',
    className: (0, _classnames["default"])(styles['EmptyState_paragraph'], styles['EmptyState_element'])
  }, descriptionProps.text), children));
}
