// @contentful/contentful-slatejs-adapter/dist/lib/__test__/contentful-helpers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Contentful = require("@contentful/rich-text-types");
function document() {
    var content = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        content[_i] = arguments[_i];
    }
    return {
        data: {},
        nodeType: Contentful.BLOCKS.DOCUMENT,
        content: content,
    };
}
exports.document = document;
function block(nodeType) {
    var content = [];
    for (var _i = 1; _i < arguments.length; _i++) {
        content[_i - 1] = arguments[_i];
    }
    return {
        nodeType: nodeType,
        content: content,
        data: {},
    };
}
exports.block = block;
function inline(nodeType) {
    var content = [];
    for (var _i = 1; _i < arguments.length; _i++) {
        content[_i - 1] = arguments[_i];
    }
    return {
        nodeType: nodeType,
        content: content,
        data: {},
    };
}
exports.inline = inline;
function text(value) {
    var marks = [];
    for (var _i = 1; _i < arguments.length; _i++) {
        marks[_i - 1] = arguments[_i];
    }
    return {
        nodeType: 'text',
        data: {},
        marks: marks,
        value: value,
    };
}
exports.text = text;
function mark(type) {
    return {
        type: type,
    };
}
exports.mark = mark;
