// @contentful/contentful-slatejs-adapter/dist/lib/__test__/slate-helpers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function document() {
    var nodes = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        nodes[_i] = arguments[_i];
    }
    return {
        object: 'document',
        data: {},
        nodes: nodes,
    };
}
exports.document = document;
function block(type, isVoid) {
    var nodes = [];
    for (var _i = 2; _i < arguments.length; _i++) {
        nodes[_i - 2] = arguments[_i];
    }
    return {
        object: 'block',
        isVoid: isVoid,
        data: {},
        type: type,
        nodes: nodes,
    };
}
exports.block = block;
function inline(type, isVoid) {
    var nodes = [];
    for (var _i = 2; _i < arguments.length; _i++) {
        nodes[_i - 2] = arguments[_i];
    }
    return {
        object: 'inline',
        isVoid: isVoid,
        data: {},
        type: type,
        nodes: nodes,
    };
}
exports.inline = inline;
function text() {
    var leaves = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        leaves[_i] = arguments[_i];
    }
    return {
        object: 'text',
        data: {},
        leaves: leaves,
    };
}
exports.text = text;
function leaf(value) {
    var marks = [];
    for (var _i = 1; _i < arguments.length; _i++) {
        marks[_i - 1] = arguments[_i];
    }
    return {
        object: 'leaf',
        text: value,
        marks: marks,
    };
}
exports.leaf = leaf;
function mark(type, data) {
    return {
        type: type,
        data: data || {},
        object: 'mark',
    };
}
exports.mark = mark;
