// @contentful/forma-36-react-components/dist/components/Switch/Switch.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Switch = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Icon = require("../Icon");

var styles = {
  "Switch__wrapper": "Switch__Switch__wrapper___1kkJg",
  "Switch": "Switch__Switch___bJH7R",
  "Switch--small": "Switch__Switch--small___hUOJI",
  "Switch__label": "Switch__Switch__label___2S5bb",
  "Switch__icon--done": "Switch__Switch__icon--done___uauUi",
  "Switch__icon--close": "Switch__Switch__icon--close___2kKlc",
  "Switch__icon": "Switch__Switch__icon___18JVx",
  "Switch--checked": "Switch__Switch--checked___3CVQe",
  "Switch--disabled": "Switch__Switch--disabled___4AE9X",
  "Switch__label--disabled": "Switch__Switch__label--disabled___2C2Sn"
};

var Switch = function Switch(props) {
  var _classNames;

  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(styles['Switch__wrapper'], {}, props.className)
  }, /*#__PURE__*/_react["default"].createElement("input", {
    type: "checkbox",
    onClick: onToggle,
    onKeyUp: onKeyUp,
    defaultChecked: props.isChecked,
    disabled: props.isDisabled,
    className: (0, _classnames["default"])(styles['Switch'], (_classNames = {}, (0, _defineProperty2["default"])(_classNames, styles['Switch--checked'], props.isChecked), (0, _defineProperty2["default"])(_classNames, styles['Switch--disabled'], props.isDisabled), (0, _defineProperty2["default"])(_classNames, styles['Switch--small'], props.size === 'small'), _classNames)),
    id: props.id
  }), /*#__PURE__*/_react["default"].createElement("label", {
    className: (0, _classnames["default"])(styles['Switch__label'], (0, _defineProperty2["default"])({}, styles['Switch__label--disabled'], props.isDisabled)),
    htmlFor: props.id
  }, /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: "Done",
    color: "white",
    size: "tiny",
    className: (0, _classnames["default"])(styles['Switch__icon'], styles['Switch__icon--done'])
  }), props.labelText, /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: "Close",
    color: "white",
    size: "tiny",
    className: (0, _classnames["default"])(styles['Switch__icon'], styles['Switch__icon--close'])
  })));

  function onToggle() {
    if (props.onToggle) {
      props.onToggle(!props.isChecked);
    }
  }

  function onKeyUp(e) {
    if (props.onToggle && e.key === 'Enter') {
      props.onToggle(!props.isChecked);
    }
  }
};

exports.Switch = Switch;
