// @contentful/forma-36-react-components/dist/components/DateTime/dateUtils.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.formatWeekdayDate = exports.formatTime = exports.formatRelativeDateTime = exports.formatMachineReadableDateTime = exports.formatDateTime = exports.formatDate = void 0;

var _dayjs = _interopRequireDefault(require("dayjs"));

var _relativeTime = _interopRequireDefault(require("dayjs/plugin/relativeTime"));

var _utc = _interopRequireDefault(require("dayjs/plugin/utc"));

_dayjs["default"].extend(_utc["default"]);

_dayjs["default"].extend(_relativeTime["default"]);

var formatTokens = {
  FULL: 'D MMM YYYY[ at ]h:mm a',
  DATE_ONLY: 'D MMM YYYY',
  TIME_ONLY: 'h:mm a',
  WEEKDAY_DATE: 'ddd[, ]D MMM'
};
var formatTokensMachineReadable = {
  FULL: 'YYYY-MM-DDTHH:mm:ss.SSS[Z]',
  DATE_ONLY: 'YYYY-MM-DD',
  TIME_ONLY: 'HH:mm:ss.SSS',
  WEEKDAY_DATE: 'MM-DD'
};

/**
 * @example
 * > formatDateTime(date)
 * 13 Aug 2019 at 10:00 am
 *
 * @example
 * > formatDateTime(date, 'DATE_ONLY')
 * 13 Aug 2019
 */
var formatDateTime = function formatDateTime(date) {
  var token = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'FULL';

  if (!formatTokens[token]) {
    throw new TypeError("Unknown date format '".concat(token, "'"));
  }

  return (0, _dayjs["default"])(date).format(formatTokens[token]);
};
/**
 * @example
 * > formatMachineReadableDateTime(date)
 * 2019-08-13T10:00:00.000Z
 *
 * @example
 * > formatMachineReadableDateTime(date, 'DATE_ONLY')
 * 2019-08-13
 */


exports.formatDateTime = formatDateTime;

var formatMachineReadableDateTime = function formatMachineReadableDateTime(date) {
  var token = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'FULL';

  if (!formatTokens[token]) {
    throw new TypeError("Unknown date format '".concat(token, "'"));
  }

  return (0, _dayjs["default"])(date).utc().format(formatTokensMachineReadable[token]);
};
/**
 * @example
 * > formatDate(date)
 * 13 Aug 2019
 */


exports.formatMachineReadableDateTime = formatMachineReadableDateTime;

var formatDate = function formatDate(date) {
  return formatDateTime(date, 'DATE_ONLY');
};
/**
 * @example
 * > formatTime(date)
 * 8:00 am
 */


exports.formatDate = formatDate;

var formatTime = function formatTime(date) {
  return formatDateTime(date, 'TIME_ONLY');
};
/**
 * @example
 * > formatWeekdayDate(date)
 * Mon, 12 Aug
 */


exports.formatTime = formatTime;

var formatWeekdayDate = function formatWeekdayDate(date) {
  return formatDateTime(date, 'WEEKDAY_DATE');
};

exports.formatWeekdayDate = formatWeekdayDate;

var formatRelativeDateTime = function formatRelativeDateTime(date) {
  var baseDate = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : new Date();
  return (0, _dayjs["default"])(date).from(baseDate);
};

exports.formatRelativeDateTime = formatRelativeDateTime;
