// @contentful/forma-36-react-components/dist/components/Card/EntryCard/EntryCard.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EntryCard = EntryCard;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _truncate = _interopRequireDefault(require("truncate"));

var _Card = require("../Card");

var _CardActions = require("../CardActions/CardActions");

var _Tag = require("../../Tag");

var _EntryCardSkeleton = require("./EntryCardSkeleton");

var _CardDragHandle = require("../CardDragHandle/CardDragHandle");

var _Icon = require("../../Icon");

var _excluded = ["className", "title", "onClick", "description", "contentType", "status", "statusIcon", "thumbnailElement", "loading", "dropdownListElements", "isDragActive", "size", "testId", "cardDragHandleComponent", "cardDragHandleProps", "withDragHandle"];
var styles = {
  "EntryCard": "EntryCard__EntryCard___2kIVv",
  "EntryCard__header": "EntryCard__EntryCard__header___3s_4z",
  "EntryCard--size-default": "EntryCard__EntryCard--size-default___ahhv9",
  "EntryCard--size-small": "EntryCard__EntryCard--size-small___wmiCz",
  "EntryCard--size-auto": "EntryCard__EntryCard--size-auto___roGvN",
  "EntryCard__wrapper": "EntryCard__EntryCard__wrapper___2i20k",
  "EntryCard__skeleton-wrapper": "EntryCard__EntryCard__skeleton-wrapper___1p3KI",
  "EntryCard--drag-active": "EntryCard__EntryCard--drag-active___26ZqT",
  "EntryCard--is-loading": "EntryCard__EntryCard--is-loading___310RH",
  "EntryCard__content-type": "EntryCard__EntryCard__content-type___77aij",
  "EntryCard__icon": "EntryCard__EntryCard__icon___27uFX",
  "EntryCard__actions": "EntryCard__EntryCard__actions___1yMP3",
  "EntryCard__content": "EntryCard__EntryCard__content___1pcqO",
  "EntryCard__body": "EntryCard__EntryCard__body___21HhK",
  "EntryCard__title": "EntryCard__EntryCard__title___2q3bn",
  "EntryCard__description": "EntryCard__EntryCard__description___11s4-",
  "EntryCard__thumbnail": "EntryCard__EntryCard__thumbnail___3mEp5"
};

function EntryCard(_ref) {
  var className = _ref.className,
      _ref$title = _ref.title,
      title = _ref$title === void 0 ? 'Untitled' : _ref$title,
      onClick = _ref.onClick,
      description = _ref.description,
      contentType = _ref.contentType,
      status = _ref.status,
      statusIcon = _ref.statusIcon,
      thumbnailElement = _ref.thumbnailElement,
      loading = _ref.loading,
      dropdownListElements = _ref.dropdownListElements,
      isDragActive = _ref.isDragActive,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'default' : _ref$size,
      _ref$testId = _ref.testId,
      testId = _ref$testId === void 0 ? 'cf-ui-entry-card' : _ref$testId,
      cardDragHandleComponent = _ref.cardDragHandleComponent,
      cardDragHandleProps = _ref.cardDragHandleProps,
      withDragHandle = _ref.withDragHandle,
      otherProps = (0, _objectWithoutProperties2["default"])(_ref, _excluded);
  var renderTitle = (0, _react.useCallback)(function (_size, title) {
    if (!title) {
      return;
    }

    var truncatedTitle = (0, _truncate["default"])(title, 255, {});
    return /*#__PURE__*/_react["default"].createElement("h1", {
      title: title.length > 255 ? title : '',
      className: styles.EntryCard__title,
      "data-test-id": "title"
    }, truncatedTitle);
  }, []);
  var renderDescription = (0, _react.useCallback)(function (size, description) {
    if (!description || size === 'small') {
      return;
    }

    var truncatedDescription = (0, _truncate["default"])(description, 95, {});
    return /*#__PURE__*/_react["default"].createElement("p", {
      className: styles.EntryCard__description
    }, truncatedDescription);
  }, []);
  var renderThumbnail = (0, _react.useCallback)(function (size, thumbnailElement) {
    if (!thumbnailElement || size === 'small') {
      return;
    }

    return /*#__PURE__*/_react["default"].createElement("figure", {
      className: styles.EntryCard__thumbnail
    }, thumbnailElement);
  }, []);
  var renderStatus = (0, _react.useCallback)(function (status) {
    var label;
    var type;

    switch (status) {
      case 'archived':
        label = 'archived';
        type = 'negative';
        break;

      case 'changed':
        label = 'changed';
        type = 'primary';
        break;

      case 'published':
        label = 'published';
        type = 'positive';
        break;

      default:
        label = 'draft';
        type = 'warning';
    }

    return /*#__PURE__*/_react["default"].createElement(_Tag.Tag, {
      tagType: type
    }, label);
  }, []);
  var renderCardDragHandle = (0, _react.useCallback)(function () {
    if (cardDragHandleComponent) {
      return cardDragHandleComponent;
    } else if (withDragHandle) {
      return /*#__PURE__*/_react["default"].createElement(_CardDragHandle.CardDragHandle, (0, _extends2["default"])({
        isDragActive: isDragActive
      }, cardDragHandleProps), "Reorder entry");
    }
  }, [cardDragHandleComponent, cardDragHandleProps, isDragActive, withDragHandle]);
  var classNames = (0, _classnames["default"])(styles.EntryCard, styles["EntryCard--size-".concat(size)], (0, _defineProperty2["default"])({}, styles['EntryCard--drag-active'], isDragActive), className);
  return /*#__PURE__*/_react["default"].createElement(_Card.Card, (0, _extends2["default"])({
    className: classNames,
    onClick: !loading ? onClick : undefined,
    padding: "none",
    "data-test-id": testId
  }, otherProps), loading ? /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(styles.EntryCard__wrapper, styles['EntryCard__skeleton-wrapper'])
  }, /*#__PURE__*/_react["default"].createElement(_EntryCardSkeleton.EntryCardSkeleton, null)) : /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, renderCardDragHandle(), /*#__PURE__*/_react["default"].createElement("div", {
    className: styles.EntryCard__wrapper
  }, /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("div", {
    className: styles.EntryCard__header
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: styles['EntryCard__content-type'],
    "data-test-id": "content-type"
  }, contentType), statusIcon && typeof statusIcon === 'string' ? /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: statusIcon,
    color: "muted",
    className: styles['EntryCard__icon']
  }) : statusIcon, status && renderStatus(status), dropdownListElements && /*#__PURE__*/_react["default"].createElement(_CardActions.CardActions, {
    className: styles['EntryCard__actions'],
    iconButtonProps: {
      onClick: function onClick(e) {
        return e.stopPropagation;
      }
    }
  }, dropdownListElements)), /*#__PURE__*/_react["default"].createElement("div", {
    className: styles.EntryCard__content
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: styles.EntryCard__body
  }, renderTitle(size, title), renderDescription(size, description)), renderThumbnail(size, thumbnailElement))))));
}
