// @contentful/forma-36-react-components/dist/components/Asset/Asset.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Asset = Asset;
exports.isAssetType = isAssetType;
exports.types = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _AssetIcon = require("./AssetIcon/AssetIcon");

var _excluded = ["className", "src", "status", "testId", "title", "type"];
var styles = {
  "Asset": "Asset__Asset___1zgnB",
  "Asset__image-container": "Asset__Asset__image-container___1oHDv",
  "Asset__image-container__image": "Asset__Asset__image-container__image___3J2Ik",
  "Asset__title-container": "Asset__Asset__title-container___jUj2R",
  "Asset__title-container__title": "Asset__Asset__title-container__title___1AHiK",
  "Asset__asset-container": "Asset__Asset__asset-container___226So",
  "Asset__asset-container__title": "Asset__Asset__asset-container__title___y3fAq",
  "Asset__illustration-container": "Asset__Asset__illustration-container___D0AVE"
};
var types = {
  archive: 'archive',
  audio: 'audio',
  code: 'code',
  image: 'image',
  markup: 'markup',
  pdf: 'pdf',
  plaintext: 'plaintext',
  presentation: 'presentation',
  richtext: 'richtext',
  spreadsheet: 'spreadsheet',
  video: 'video'
};
exports.types = types;

function isAssetType(type) {
  return Object.keys(types).includes(type);
}

function Asset(_ref) {
  var className = _ref.className,
      src = _ref.src,
      status = _ref.status,
      _ref$testId = _ref.testId,
      testId = _ref$testId === void 0 ? 'cf-ui-asset' : _ref$testId,
      title = _ref.title,
      _ref$type = _ref.type,
      type = _ref$type === void 0 ? 'image' : _ref$type,
      otherProps = (0, _objectWithoutProperties2["default"])(_ref, _excluded);
  var classNames = (0, _classnames["default"])(styles.Asset, className);
  var isImage = type === 'image' && src !== undefined && src !== ''; // Do not show image previews when publish status is archived

  var showPreview = isImage && status !== 'archived';
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({
    className: classNames,
    "data-test-id": testId
  }, otherProps), showPreview ? /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("div", {
    className: styles['Asset__image-container']
  }, /*#__PURE__*/_react["default"].createElement("img", {
    className: styles['Asset__image-container__image'],
    src: src,
    alt: title
  })), title && /*#__PURE__*/_react["default"].createElement("div", {
    className: styles['Asset__title-container']
  }, /*#__PURE__*/_react["default"].createElement("span", {
    className: styles['Asset__title-container__title']
  }, title))) : /*#__PURE__*/_react["default"].createElement("div", {
    className: styles['Asset__asset-container']
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: styles['Asset__illustration-container']
  }, /*#__PURE__*/_react["default"].createElement(_AssetIcon.AssetIcon, {
    type: type
  })), title && /*#__PURE__*/_react["default"].createElement("span", {
    className: styles['Asset__asset-container__title']
  }, title)));
}
