// @contentful/forma-36-react-components/dist/components/Illustration/Illustration.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Illustration = Illustration;
exports["default"] = void 0;
exports.isIllustrationType = isIllustrationType;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _constants = require("./constants");

var _svg = require("./svg");

var _excluded = ["className", "testId", "illustration"];
var styles = {
  "Illustration": "Illustration__Illustration___1R5Px"
};
var ILLUSTRATION_NAMES = Object.keys(_constants.illustrationName);

function isIllustrationType(name) {
  return ILLUSTRATION_NAMES.includes(name);
}

function Illustration(_ref) {
  var className = _ref.className,
      _ref$testId = _ref.testId,
      testId = _ref$testId === void 0 ? 'cf-ui-illustration' : _ref$testId,
      illustration = _ref.illustration,
      otherProps = (0, _objectWithoutProperties2["default"])(_ref, _excluded);
  var illustrationComponents = {
    Archive: _svg.Archive,
    Audio: _svg.Audio,
    Richtext: _svg.Richtext,
    Code: _svg.Code,
    Image: _svg.Image,
    Markup: _svg.Markup,
    Pdf: _svg.Pdf,
    Plaintext: _svg.Plaintext,
    Presentation: _svg.Presentation,
    Spreadsheet: _svg.Spreadsheet,
    Video: _svg.Video
  };
  var classNames = (0, _classnames["default"])(styles['Illustration'], className);
  var Element = illustrationComponents[illustration];
  return /*#__PURE__*/_react["default"].createElement(Element, (0, _extends2["default"])({}, otherProps, {
    "data-test-id": testId,
    className: classNames
  }));
}

var _default = Illustration;
exports["default"] = _default;
