// @contentful/forma-36-react-components/dist/components/Card/AssetCard/AssetCard.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AssetCard = AssetCard;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Card = require("../Card");

var _Icon = require("../../Icon");

var _CardActions = require("../CardActions/CardActions");

var _Asset = require("../../Asset");

var _Tag = require("../../Tag");

var _AssetCardSkeleton = require("./AssetCardSkeleton");

var _CardDragHandle = require("../CardDragHandle/CardDragHandle");

var _excluded = ["className", "src", "type", "title", "status", "statusIcon", "isLoading", "dropdownListElements", "isDragActive", "size", "testId", "cardDragHandleProps", "cardDragHandleComponent", "withDragHandle"];
var styles = {
  "AssetCard": "AssetCard__AssetCard____VWXj",
  "AssetCard--size-small": "AssetCard__AssetCard--size-small___2fyHk",
  "AssetCard__asset": "AssetCard__AssetCard__asset___189id",
  "AssetCard--drag-active": "AssetCard__AssetCard--drag-active___WtiNB",
  "AssetCard__wrapper": "AssetCard__AssetCard__wrapper___24k3w",
  "AssetCard__content": "AssetCard__AssetCard__content___2ZY73",
  "AssetCard__header": "AssetCard__AssetCard__header___2ahyT",
  "AssetCard__icon": "AssetCard__AssetCard__icon___1rpC0",
  "AssetCard__actions": "AssetCard__AssetCard__actions___2Nbjg"
};

function AssetCard(_ref) {
  var _cn;

  var className = _ref.className,
      src = _ref.src,
      type = _ref.type,
      title = _ref.title,
      status = _ref.status,
      statusIcon = _ref.statusIcon,
      _ref$isLoading = _ref.isLoading,
      isLoading = _ref$isLoading === void 0 ? false : _ref$isLoading,
      dropdownListElements = _ref.dropdownListElements,
      isDragActive = _ref.isDragActive,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'default' : _ref$size,
      _ref$testId = _ref.testId,
      testId = _ref$testId === void 0 ? 'cf-ui-asset-card' : _ref$testId,
      cardDragHandleProps = _ref.cardDragHandleProps,
      cardDragHandleComponent = _ref.cardDragHandleComponent,
      withDragHandle = _ref.withDragHandle,
      otherProps = (0, _objectWithoutProperties2["default"])(_ref, _excluded);
  var renderStatus = (0, _react.useCallback)(function (status, statusIcon) {
    var label;
    var type = null;

    switch (status) {
      case 'archived':
        label = 'archived';
        type = 'negative';
        break;

      case 'changed':
        label = 'changed';
        type = 'primary';
        break;

      case 'published':
        label = 'published';
        type = 'positive';
        break;

      default:
        label = 'draft';
        type = 'warning';
    }

    return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, statusIcon && typeof statusIcon === 'string' ? /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
      icon: statusIcon,
      color: "muted",
      className: styles['AssetCard__icon']
    }) : statusIcon, /*#__PURE__*/_react["default"].createElement(_Tag.Tag, {
      tagType: type
    }, label));
  }, []);
  var renderCardDragHandle = (0, _react.useCallback)(function () {
    if (cardDragHandleComponent) {
      return cardDragHandleComponent;
    } else if (withDragHandle) {
      return /*#__PURE__*/_react["default"].createElement(_CardDragHandle.CardDragHandle, (0, _extends2["default"])({
        isDragActive: isDragActive
      }, cardDragHandleProps), "Reorder entry");
    }
  }, [cardDragHandleComponent, cardDragHandleProps, isDragActive, withDragHandle]);
  var classNames = (0, _classnames["default"])(styles.AssetCard, (_cn = {}, (0, _defineProperty2["default"])(_cn, styles['AssetCard--drag-active'], isDragActive), (0, _defineProperty2["default"])(_cn, styles["AssetCard--size-".concat(size)], size), _cn), className);
  return /*#__PURE__*/_react["default"].createElement(_Card.Card, (0, _extends2["default"])({
    className: classNames,
    padding: "none",
    testId: testId,
    title: title
  }, otherProps), isLoading ? /*#__PURE__*/_react["default"].createElement(_AssetCardSkeleton.AssetCardSkeleton, {
    size: size
  }) : /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, renderCardDragHandle(), /*#__PURE__*/_react["default"].createElement("div", {
    className: styles['AssetCard__wrapper']
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: styles['AssetCard__header']
  }, status && renderStatus(status, statusIcon), dropdownListElements && /*#__PURE__*/_react["default"].createElement(_CardActions.CardActions, {
    className: styles['AssetCard__actions'],
    iconButtonProps: {
      onClick: function onClick(e) {
        return e.stopPropagation;
      }
    }
  }, dropdownListElements)), /*#__PURE__*/_react["default"].createElement("div", {
    className: styles['AssetCard__content']
  }, /*#__PURE__*/_react["default"].createElement(_Asset.Asset, {
    className: styles['AssetCard__asset'],
    src: src,
    title: title,
    type: type,
    status: status
  })))));
}
