// @contentful/contentful-slatejs-adapter/dist/lib/slatejs-to-contentful-adapter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var lodash_flatmap_1 = require("lodash.flatmap");
var Contentful = require("@contentful/rich-text-types");
var helpers_1 = require("./helpers");
var schema_1 = require("./schema");
function toContentfulDocument(_a) {
    var document = _a.document, schema = _a.schema;
    return {
        nodeType: Contentful.BLOCKS.DOCUMENT,
        data: helpers_1.getDataOfDefault(document.data),
        content: lodash_flatmap_1.default(document.nodes, function (node) { return convertNode(node, schema_1.fromJSON(schema)); }),
    };
}
exports.default = toContentfulDocument;
function convertNode(node, schema) {
    var nodes = [];
    switch (node.object) {
        case 'block':
        case 'inline':
            var slateNode = node;
            var content = lodash_flatmap_1.default(slateNode.nodes, function (childNode) { return convertNode(childNode, schema); });
            if (!slateNode.type) {
                throw new Error("Unexpected slate node " + JSON.stringify(slateNode));
            }
            var contentfulBlock = {
                nodeType: slateNode.type,
                content: [],
                data: helpers_1.getDataOfDefault(slateNode.data),
            };
            if (!schema.isVoid(contentfulBlock)) {
                contentfulBlock.content = content;
            }
            nodes.push(contentfulBlock);
            break;
        case 'text':
            convertText(node).forEach(function (childNode) { return nodes.push(childNode); });
            break;
        default:
            assertUnreachable(node);
            break;
    }
    return nodes;
}
function convertText(node) {
    return node.leaves.map(function (leaf) { return ({
        nodeType: 'text',
        value: leaf.text,
        marks: leaf.marks ? leaf.marks.map(function (mark) { return ({ type: mark.type }); }) : [],
        data: helpers_1.getDataOfDefault(node.data),
    }); });
}
function assertUnreachable(object) {
    throw new Error("Unexpected slate object " + object);
}
