// @contentful/forma-36-react-components/dist/components/Tag/Tag.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Tag = Tag;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _excluded = ["className", "children", "tagType", "size", "entityStatusType", "testId"];
var styles = {
  "Tag": "Tag__Tag___Y-myd",
  "Tag--primary": "Tag__Tag--primary___2Hk3I",
  "Tag--positive": "Tag__Tag--positive___1cepi",
  "Tag--negative": "Tag__Tag--negative___12luW",
  "Tag--warning": "Tag__Tag--warning___3Bet2",
  "Tag--secondary": "Tag__Tag--secondary___2vTK0",
  "Tag--muted": "Tag__Tag--muted___1Uba5",
  "Tag--primary-filled": "Tag__Tag--primary-filled___36ZvU",
  "Tag--featured": "Tag__Tag--featured___1m-dR",
  "Tag--small": "Tag__Tag--small___xgHQ9"
};
var statusTagTypeMap = {
  published: 'positive',
  draft: 'warning',
  archived: 'negative',
  changed: 'primary',
  deleted: 'negative',
  "new": 'primary-filled'
};

function Tag(_ref) {
  var _cn;

  var className = _ref.className,
      children = _ref.children,
      _ref$tagType = _ref.tagType,
      tagType = _ref$tagType === void 0 ? 'primary' : _ref$tagType,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'default' : _ref$size,
      entityStatusType = _ref.entityStatusType,
      _ref$testId = _ref.testId,
      testId = _ref$testId === void 0 ? 'cf-ui-tag' : _ref$testId,
      otherProps = (0, _objectWithoutProperties2["default"])(_ref, _excluded);
  var classNames = (0, _classnames["default"])(styles.Tag, className, (_cn = {}, (0, _defineProperty2["default"])(_cn, styles["Tag--".concat(entityStatusType ? statusTagTypeMap[entityStatusType] : tagType)], entityStatusType ? statusTagTypeMap[entityStatusType] : tagType), (0, _defineProperty2["default"])(_cn, styles["Tag--".concat(size)], size !== 'default'), _cn));
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({
    className: classNames,
    "data-test-id": testId
  }, otherProps), children);
}
