// @contentful/forma-36-react-components/dist/components/Workbench/Workbench.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Workbench = Workbench;
exports.WorkbenchContent = WorkbenchContent;
exports.WorkbenchHeader = WorkbenchHeader;
exports.WorkbenchSidebar = WorkbenchSidebar;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Typography = require("../Typography");

var _IconButton = require("../IconButton");

var _excluded = ["children", "className", "position", "testId", "labelText"],
    _excluded2 = ["className", "children", "testId"];
var styles = {
  "Workbench": "Workbench__Workbench___THi9K",
  "Workbench__header": "Workbench__Workbench__header___2wo2E",
  "Workbench__header-back": "Workbench__Workbench__header-back___3qZkM",
  "Workbench__header-back-button": "Workbench__Workbench__header-back-button___1G3-G",
  "Workbench__header-icon": "Workbench__Workbench__header-icon___1kJlL",
  "Workbench__header-title": "Workbench__Workbench__header-title___3xp7b",
  "Workbench__header-title__heading": "Workbench__Workbench__header-title__heading___1Pe5h",
  "Workbench__header-description": "Workbench__Workbench__header-description___1LHbc",
  "Workbench__header-description__text": "Workbench__Workbench__header-description__text___2xjBG",
  "Workbench__header-actions": "Workbench__Workbench__header-actions___mJoSR",
  "Workbench__content-wrapper": "Workbench__Workbench__content-wrapper___3yWKi",
  "Workbench__content": "Workbench__Workbench__content___1U8bV",
  "Workbench__content-inner": "Workbench__Workbench__content-inner___36_Hx",
  "Workbench__content-inner--default": "Workbench__Workbench__content-inner--default___2wRXn",
  "Workbench__content-inner--text": "Workbench__Workbench__content-inner--text___7nfTX",
  "Workbench__content-inner--full": "Workbench__Workbench__content-inner--full___3eP1v",
  "Workbench__sidebar": "Workbench__Workbench__sidebar___3i7n0",
  "Workbench__sidebar--left": "Workbench__Workbench__sidebar--left___67OKM",
  "Workbench__sidebar--right": "Workbench__Workbench__sidebar--right___3f4XO"
};

function WorkbenchHeader(_ref) {
  var actions = _ref.actions,
      className = _ref.className,
      description = _ref.description,
      icon = _ref.icon,
      onBack = _ref.onBack,
      _ref$testId = _ref.testId,
      testId = _ref$testId === void 0 ? 'cf-ui-workbench-header' : _ref$testId,
      title = _ref.title;
  return /*#__PURE__*/_react["default"].createElement("header", {
    className: (0, _classnames["default"])(styles['Workbench__header'], className),
    "data-test-id": testId
  }, onBack ? /*#__PURE__*/_react["default"].createElement("div", {
    className: styles['Workbench__header-back']
  }, /*#__PURE__*/_react["default"].createElement(_IconButton.IconButton, {
    onClick: function onClick() {
      if (typeof onBack === 'function') {
        onBack();
      }
    },
    testId: "workbench-back-btn",
    className: styles['Workbench__header-back-button'],
    label: "Back",
    buttonType: "muted",
    iconProps: {
      size: 'large',
      icon: 'ChevronLeft'
    }
  })) : null, icon ? /*#__PURE__*/_react["default"].createElement("div", {
    className: styles['Workbench__header-icon']
  }, icon) : null, title && /*#__PURE__*/_react["default"].createElement("div", {
    "data-test-id": "workbench-title",
    className: styles['Workbench__header-title']
  }, typeof title === 'string' ? /*#__PURE__*/_react["default"].createElement(_Typography.Heading, {
    element: "h1",
    className: styles['Workbench__header-title__heading']
  }, title) : title), description && /*#__PURE__*/_react["default"].createElement("div", {
    "data-test-id": "workbench-description",
    className: styles['Workbench__header-description']
  }, typeof description === 'string' ? /*#__PURE__*/_react["default"].createElement("p", {
    className: styles['Workbench__header-description__text']
  }, description) : description), actions ? /*#__PURE__*/_react["default"].createElement("div", {
    className: styles['Workbench__header-actions']
  }, actions) : null);
}

WorkbenchHeader.displayName = 'Workbench.Header';

function WorkbenchSidebar(_ref2) {
  var _cn;

  var children = _ref2.children,
      className = _ref2.className,
      position = _ref2.position,
      _ref2$testId = _ref2.testId,
      testId = _ref2$testId === void 0 ? 'cf-ui-workbench-sidebar' : _ref2$testId,
      _ref2$labelText = _ref2.labelText,
      labelText = _ref2$labelText === void 0 ? position === 'right' ? 'Primary sidebar' : 'Secondary sidebar' : _ref2$labelText,
      otherProps = (0, _objectWithoutProperties2["default"])(_ref2, _excluded);
  return /*#__PURE__*/_react["default"].createElement("aside", (0, _extends2["default"])({
    "data-test-id": "".concat(testId).concat(position ? "-".concat(position) : ''),
    className: (0, _classnames["default"])(styles['Workbench__sidebar'], (_cn = {}, (0, _defineProperty2["default"])(_cn, styles['Workbench__sidebar--right'], position === 'right'), (0, _defineProperty2["default"])(_cn, styles['Workbench__sidebar--left'], position === 'left'), _cn), className),
    "aria-label": labelText
  }, otherProps), children);
}

function WorkbenchContent(_ref3) {
  var _cn2;

  var children = _ref3.children,
      className = _ref3.className,
      _ref3$testId = _ref3.testId,
      testId = _ref3$testId === void 0 ? 'cf-ui-workbench-content' : _ref3$testId,
      _ref3$type = _ref3.type,
      type = _ref3$type === void 0 ? 'default' : _ref3$type,
      _ref3$labelText = _ref3.labelText,
      labelText = _ref3$labelText === void 0 ? 'Main content' : _ref3$labelText,
      _ref3$id = _ref3.id,
      id = _ref3$id === void 0 ? 'main-content' : _ref3$id;
  return /*#__PURE__*/_react["default"].createElement("main", {
    className: (0, _classnames["default"])(styles['Workbench__content'], className),
    "data-test-id": testId,
    "aria-label": labelText,
    id: id
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(styles['Workbench__content-inner'], (_cn2 = {}, (0, _defineProperty2["default"])(_cn2, styles['Workbench__content-inner--default'], type === 'default'), (0, _defineProperty2["default"])(_cn2, styles['Workbench__content-inner--text'], type === 'text'), (0, _defineProperty2["default"])(_cn2, styles['Workbench__content-inner--full'], type === 'full'), _cn2))
  }, children));
}

function Workbench(_ref4) {
  var className = _ref4.className,
      children = _ref4.children,
      _ref4$testId = _ref4.testId,
      testId = _ref4$testId === void 0 ? 'cf-ui-workbench' : _ref4$testId,
      otherProps = (0, _objectWithoutProperties2["default"])(_ref4, _excluded2);

  var childrenArray = _react["default"].Children.toArray(children);

  var header = [];
  var other = [];
  childrenArray.forEach( // eslint-disable-next-line
  function (item) {
    if (item && item.type && item.type.displayName === 'Workbench.Header') {
      header.push(item);
    } else {
      other.push(item);
    }
  });
  var classNames = (0, _classnames["default"])(styles.Workbench, className);
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({}, otherProps, {
    className: classNames,
    "data-test-id": testId
  }), header, /*#__PURE__*/_react["default"].createElement("div", {
    className: styles['Workbench__content-wrapper']
  }, other));
}

Workbench.Header = WorkbenchHeader;
Workbench.Content = WorkbenchContent;
Workbench.Sidebar = WorkbenchSidebar;
