// @contentful/forma-36-react-components/dist/components/Dropdown/Dropdown.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Dropdown = Dropdown;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _reactPopper = require("react-popper");

var _DropdownListItem = require("./DropdownListItem/DropdownListItem");

var _DropdownContainer = require("./DropdownContainer/DropdownContainer");

var _excluded = ["children", "className", "dropdownContainerClassName", "getContainerRef", "isAutoalignmentEnabled", "isFullWidth", "isOpen", "onClose", "position", "submenuToggleLabel", "testId", "toggleElement", "usePortal", "nonClosingRefs", "focusContainerOnOpen"];
var styles = {
  "Dropdown": "Dropdown__Dropdown___nAsJ-"
};
var offsetModifier = {
  name: 'offset',
  options: {
    offset: [0, 0]
  }
};
/**
 * Popper.js modifier to give the popper element the full width of the reference
 */

var sameWidth = {
  name: 'sameWidth',
  enabled: true,
  phase: 'beforeWrite',
  requires: ['computeStyles'],
  fn: function fn(_ref) {
    var state = _ref.state;
    state.styles.popper.width = "".concat(state.rects.reference.width, "px");
  },
  effect: function effect(_ref2) {
    var state = _ref2.state;
    var reference = state.elements.reference;
    state.elements.popper.style.width = "".concat(reference.offsetWidth, "px");
    return function () {};
  }
};

/**
 * Helper method to map our current Dropdown position props to Popper.js
 * placements.
 *
 * @todo: Maybe we can use the Popper placements in the next breaking change?
 */
var mapPositionTypeToPlacement = function mapPositionTypeToPlacement(position) {
  switch (position) {
    case 'bottom-left':
      return 'bottom-start';

    case 'bottom-right':
      return 'bottom-end';

    case 'right':
      return 'right-start';

    case 'left':
      return 'left-start';

    case 'top-left':
      return 'top-start';

    case 'top-right':
      return 'top-end';

    default:
      return position;
  }
};

function Dropdown(_ref3) {
  var children = _ref3.children,
      className = _ref3.className,
      dropdownContainerClassName = _ref3.dropdownContainerClassName,
      getContainerRef = _ref3.getContainerRef,
      _ref3$isAutoalignment = _ref3.isAutoalignmentEnabled,
      isAutoalignmentEnabled = _ref3$isAutoalignment === void 0 ? true : _ref3$isAutoalignment,
      isFullWidth = _ref3.isFullWidth,
      _ref3$isOpen = _ref3.isOpen,
      isOpenProp = _ref3$isOpen === void 0 ? false : _ref3$isOpen,
      onClose = _ref3.onClose,
      _ref3$position = _ref3.position,
      position = _ref3$position === void 0 ? 'bottom-left' : _ref3$position,
      submenuToggleLabel = _ref3.submenuToggleLabel,
      _ref3$testId = _ref3.testId,
      testId = _ref3$testId === void 0 ? 'cf-ui-dropdown' : _ref3$testId,
      toggleElement = _ref3.toggleElement,
      usePortal = _ref3.usePortal,
      nonClosingRefs = _ref3.nonClosingRefs,
      _ref3$focusContainerO = _ref3.focusContainerOnOpen,
      focusContainerOnOpen = _ref3$focusContainerO === void 0 ? true : _ref3$focusContainerO,
      otherProps = (0, _objectWithoutProperties2["default"])(_ref3, _excluded);

  var _useState = (0, _react.useState)(null),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      referenceElement = _useState2[0],
      setReferenceElement = _useState2[1];

  var _useState3 = (0, _react.useState)(null),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      popperElement = _useState4[0],
      setPopperElement = _useState4[1];

  var _useState5 = (0, _react.useState)(isOpenProp),
      _useState6 = (0, _slicedToArray2["default"])(_useState5, 2),
      isOpen = _useState6[0],
      setIsOpen = _useState6[1];

  var placement = mapPositionTypeToPlacement(position);

  var _usePopper = (0, _reactPopper.usePopper)(referenceElement, popperElement, {
    placement: placement,
    modifiers: [offsetModifier, isFullWidth ? sameWidth : {}, isAutoalignmentEnabled !== undefined ? {
      name: 'preventOverflow',
      options: {
        mainAxis: isAutoalignmentEnabled
      }
    } : {}, isAutoalignmentEnabled === false ? {
      name: 'flip',
      options: {
        fallbackPlacements: []
      }
    } : {}]
  }),
      attributes = _usePopper.attributes,
      forceUpdate = _usePopper.forceUpdate,
      popperStyles = _usePopper.styles;

  var classNames = (0, _classnames["default"])(styles['Dropdown'], className);
  var containerTestId = testId ? "".concat(testId, "-container") : testId;
  var toggleElementIdSuffix = (0, _react.useRef)(Math.random() * Date.now());
  var toggleElementId = "dropdown-trigger-".concat(toggleElementIdSuffix.current);
  (0, _react.useEffect)(function () {
    setIsOpen(isOpenProp);
  }, [isOpenProp]);
  (0, _react.useEffect)(function () {
    if (forceUpdate) {
      forceUpdate();
    }
  }, [children, forceUpdate]);

  var openSubmenu = function openSubmenu(isOpen) {
    if (submenuToggleLabel) {
      setIsOpen(isOpen);
    }
  };

  var handleOnClose = (0, _react.useCallback)(function () {
    var toggleElementNode = document.querySelector("[data-node-id=\"".concat(toggleElementId, "\"]"));
    toggleElementNode === null || toggleElementNode === void 0 ? void 0 : toggleElementNode.focus();

    if (onClose) {
      onClose();
    }
  }, [toggleElementId, onClose]);
  var close = (0, _react.useCallback)(function () {
    setIsOpen(false);
    handleOnClose();
  }, [handleOnClose, setIsOpen]);
  var handleEscapeKey = (0, _react.useCallback)(function (event) {
    if (event.code === 'Escape') {
      event.stopPropagation();
      close();
    }
  }, [close]);
  (0, _react.useEffect)(function () {
    if (isOpen) {
      document.addEventListener('keydown', handleEscapeKey, true);
      return function () {
        document.removeEventListener('keydown', handleEscapeKey, true);
      };
    }
  }, [handleEscapeKey, isOpen]);
  return submenuToggleLabel ? /*#__PURE__*/_react["default"].createElement(_DropdownListItem.DropdownListItem, (0, _extends2["default"])({
    testId: testId,
    submenuToggleLabel: submenuToggleLabel,
    onEnter: function onEnter() {
      return setIsOpen(true);
    },
    onClick: function onClick() {
      return setIsOpen(true);
    },
    onLeave: function onLeave() {
      return setIsOpen(false);
    },
    "aria-expanded": isOpen,
    ref: setReferenceElement
  }, otherProps), toggleElement && /*#__PURE__*/_react["default"].cloneElement(toggleElement, {
    'aria-haspopup': 'menu',
    'aria-expanded': isOpen
  }), isOpen && /*#__PURE__*/_react["default"].createElement(_DropdownContainer.DropdownContainer, (0, _extends2["default"])({
    nonClosingRefs: nonClosingRefs,
    className: dropdownContainerClassName,
    getRef: getContainerRef,
    isOpen: isOpen,
    onClose: handleOnClose,
    openSubmenu: openSubmenu,
    ref: setPopperElement,
    style: popperStyles.popper,
    submenu: true,
    focusContainerOnOpen: focusContainerOnOpen,
    usePortal: usePortal
  }, attributes.popper), children)) : /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({
    "data-test-id": testId,
    className: classNames,
    ref: setReferenceElement
  }, otherProps), toggleElement && /*#__PURE__*/_react["default"].cloneElement(toggleElement, {
    'aria-haspopup': 'menu',
    'aria-expanded': isOpen,
    'data-node-id': toggleElementId
  }), isOpen && /*#__PURE__*/_react["default"].createElement(_DropdownContainer.DropdownContainer, (0, _extends2["default"])({
    nonClosingRefs: nonClosingRefs,
    className: dropdownContainerClassName,
    getRef: getContainerRef,
    isOpen: isOpen,
    onClose: handleOnClose,
    openSubmenu: openSubmenu,
    ref: setPopperElement,
    style: popperStyles.popper,
    submenu: false,
    focusContainerOnOpen: focusContainerOnOpen,
    testId: containerTestId,
    usePortal: usePortal
  }, attributes.popper), children));
}
