// @contentful/forma-36-react-components/dist/components/Notification/NotificationItem.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NotificationItem = NotificationItem;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _IconButton = require("../IconButton");

var _Icon = require("../Icon");

var _TextLink = require("../TextLink");

var styles = {
  "NotificationItem": "NotificationItem__NotificationItem___22iZo",
  "NotificationItem--success": "NotificationItem__NotificationItem--success___1CO7O",
  "NotificationItem--error": "NotificationItem__NotificationItem--error___19wsc",
  "NotificationItem--warning": "NotificationItem__NotificationItem--warning___Bi-Iv",
  "NotificationItem__intent": "NotificationItem__NotificationItem__intent___cC494 helpers__sr-only___3Kv3z",
  "NotificationItem__icon": "NotificationItem__NotificationItem__icon___3gdKj",
  "NotificationItem__content": "NotificationItem__NotificationItem__content___2z-pT",
  "NotificationItem__text": "NotificationItem__NotificationItem__text___1-1Up",
  "NotificationItem__title": "NotificationItem__NotificationItem__title___126c1",
  "NotificationItem__body": "NotificationItem__NotificationItem__body___11cuA",
  "NotificationItem__body--naked": "NotificationItem__NotificationItem__body--naked___2CDGp",
  "NotificationItem__dismiss": "NotificationItem__NotificationItem__dismiss___1Z6Df"
};

function NotificationItem(_ref) {
  var children = _ref.children,
      cta = _ref.cta,
      _ref$hasCloseButton = _ref.hasCloseButton,
      hasCloseButton = _ref$hasCloseButton === void 0 ? true : _ref$hasCloseButton,
      _ref$intent = _ref.intent,
      intent = _ref$intent === void 0 ? 'success' : _ref$intent,
      onClose = _ref.onClose,
      _ref$testId = _ref.testId,
      testId = _ref$testId === void 0 ? 'cf-ui-notification' : _ref$testId,
      title = _ref.title;
  var renderTitle = (0, _react.useCallback)(function () {
    return /*#__PURE__*/_react["default"].createElement("h2", {
      className: styles.NotificationItem__title
    }, title ? title : children);
  }, [children, title]);
  var renderBody = (0, _react.useCallback)(function () {
    var bodyClasses = (0, _classnames["default"])(styles.NotificationItem__body, (0, _defineProperty2["default"])({}, styles["NotificationItem__body--naked"], !title && !cta));
    return /*#__PURE__*/_react["default"].createElement("p", {
      className: bodyClasses
    }, title && children);
  }, [children, title, cta]);
  var renderCta = (0, _react.useCallback)(function () {
    if (cta && cta.label) return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(_TextLink.TextLink, cta.textLinkProps, cta.label));
  }, [cta]);
  var renderIcon = (0, _react.useCallback)(function () {
    var iconClasses = (0, _classnames["default"])(styles.NotificationItem__icon, (0, _defineProperty2["default"])({}, styles["NotificationItem__icon--".concat(intent)], true));
    var icon;
    var color;

    switch (intent) {
      case 'success':
        icon = 'CheckCircle';
        color = 'positive';
        break;

      case 'warning':
        icon = 'Warning';
        color = 'warning';
        break;

      case 'error':
        icon = 'ErrorCircle';
        color = 'negative';
        break;

      default:
        icon = 'ErrorCircle';
    }

    return /*#__PURE__*/_react["default"].createElement("div", {
      className: iconClasses,
      "aria-hidden": "true"
    }, /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
      icon: icon,
      color: color
    }));
  }, [intent]);
  var NotificationClassName = (0, _classnames["default"])(styles.NotificationItem, (0, _defineProperty2["default"])({}, styles["NotificationItem--".concat(intent)], intent));
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: NotificationClassName,
    "data-test-id": testId,
    "data-intent": intent,
    role: "alert",
    "aria-live": intent === 'success' ? 'polite' : 'assertive'
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: styles.NotificationItem__intent
  }, intent), renderIcon(), /*#__PURE__*/_react["default"].createElement("div", {
    className: styles.NotificationItem__content
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: styles.NotificationItem__text
  }, renderTitle(), renderBody(), renderCta()), hasCloseButton && /*#__PURE__*/_react["default"].createElement(_IconButton.IconButton, {
    buttonType: "secondary",
    iconProps: {
      icon: 'Close'
    },
    onClick: function onClick() {
      if (onClose) {
        onClose();
      }
    },
    testId: "cf-ui-notification-close",
    label: "Dismiss",
    className: styles.NotificationItem__dismiss
  })));
}
