// @contentful/forma-36-react-components/dist/components/Modal/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Modal", {
  enumerable: true,
  get: function get() {
    return _Modal.Modal;
  }
});
Object.defineProperty(exports, "ModalConfirm", {
  enumerable: true,
  get: function get() {
    return _ModalConfirm.ModalConfirm;
  }
});
Object.defineProperty(exports, "ModalContent", {
  enumerable: true,
  get: function get() {
    return _ModalContent.ModalContent;
  }
});
Object.defineProperty(exports, "ModalControls", {
  enumerable: true,
  get: function get() {
    return _ModalControls.ModalControls;
  }
});
Object.defineProperty(exports, "ModalHeader", {
  enumerable: true,
  get: function get() {
    return _ModalHeader.ModalHeader;
  }
});
Object.defineProperty(exports, "ModalLauncher", {
  enumerable: true,
  get: function get() {
    return _ModalLauncher.ModalLauncher;
  }
});

var _Modal = require("./Modal");

var _ModalConfirm = require("./ModalConfirm/ModalConfirm");

var _ModalContent = require("./ModalContent/ModalContent");

var _ModalControls = require("./ModalControls/ModalControls");

var _ModalHeader = require("./ModalHeader/ModalHeader");

var _ModalLauncher = require("./ModalLauncher/ModalLauncher");
