// @contentful/forma-36-react-components/dist/components/Flex/Flex.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Flex = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _forma36Tokens = _interopRequireDefault(require("@contentful/forma-36-tokens"));

var _classnames = _interopRequireDefault(require("classnames"));

var _excluded = ["alignItems", "alignSelf", "children", "className", "flexDirection", "flexGrow", "flexWrap", "fullHeight", "fullWidth", "htmlTag", "inlineFlex", "justifyContent", "justifyItems", "justifySelf", "margin", "marginBottom", "marginLeft", "marginRight", "marginTop", "noShrink", "padding", "paddingBottom", "paddingLeft", "paddingRight", "paddingTop", "style", "testId"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var styles = {
  "Flex": "Flex__Flex___Cpju1",
  "Flex__inline": "Flex__Flex__inline___39eAP",
  "Flex__fullWidth": "Flex__Flex__fullWidth___2JBEp",
  "Flex__fullHeight": "Flex__Flex__fullHeight___xgjho",
  "Flex__noShrink": "Flex__Flex__noShrink___2H1Bm"
};

var handleSpacing = function handleSpacing(value) {
  return value === 'none' ? 0 : _forma36Tokens["default"][value];
};

var Flex = function Flex(_ref) {
  var _cn;

  var alignItems = _ref.alignItems,
      alignSelf = _ref.alignSelf,
      children = _ref.children,
      className = _ref.className,
      flexDirection = _ref.flexDirection,
      flexGrow = _ref.flexGrow,
      flexWrap = _ref.flexWrap,
      fullHeight = _ref.fullHeight,
      fullWidth = _ref.fullWidth,
      _ref$htmlTag = _ref.htmlTag,
      Container = _ref$htmlTag === void 0 ? 'div' : _ref$htmlTag,
      inlineFlex = _ref.inlineFlex,
      justifyContent = _ref.justifyContent,
      justifyItems = _ref.justifyItems,
      justifySelf = _ref.justifySelf,
      margin = _ref.margin,
      marginBottom = _ref.marginBottom,
      marginLeft = _ref.marginLeft,
      marginRight = _ref.marginRight,
      marginTop = _ref.marginTop,
      noShrink = _ref.noShrink,
      padding = _ref.padding,
      paddingBottom = _ref.paddingBottom,
      paddingLeft = _ref.paddingLeft,
      paddingRight = _ref.paddingRight,
      paddingTop = _ref.paddingTop,
      style = _ref.style,
      _ref$testId = _ref.testId,
      testId = _ref$testId === void 0 ? 'cf-ui-flex' : _ref$testId,
      otherProps = (0, _objectWithoutProperties2["default"])(_ref, _excluded);
  var fullMargins = {
    margin: margin && handleSpacing(margin)
  };
  var sidesMargins = {
    marginTop: marginTop && handleSpacing(marginTop),
    marginRight: marginRight && handleSpacing(marginRight),
    marginBottom: marginBottom && handleSpacing(marginBottom),
    marginLeft: marginLeft && handleSpacing(marginLeft)
  };
  var fullPaddings = {
    padding: padding && handleSpacing(padding)
  };
  var sidesPaddings = {
    paddingTop: paddingTop && handleSpacing(paddingTop),
    paddingRight: paddingRight && handleSpacing(paddingRight),
    paddingBottom: paddingBottom && handleSpacing(paddingBottom),
    paddingLeft: paddingLeft && handleSpacing(paddingLeft)
  };
  var marginResult = margin ? fullMargins : sidesMargins;
  var paddingsResult = padding ? fullPaddings : sidesPaddings;
  var inlineStyle = {
    flexDirection: flexDirection,
    justifyContent: justifyContent,
    justifyItems: justifyItems,
    justifySelf: justifySelf,
    alignItems: alignItems,
    alignSelf: alignSelf,
    flexWrap: flexWrap,
    flexGrow: flexGrow
  };
  var classNames = (0, _classnames["default"])(styles.Flex, className, (_cn = {}, (0, _defineProperty2["default"])(_cn, styles.Flex__inline, inlineFlex), (0, _defineProperty2["default"])(_cn, styles.Flex__fullWidth, fullWidth), (0, _defineProperty2["default"])(_cn, styles.Flex__fullHeight, fullHeight), (0, _defineProperty2["default"])(_cn, styles.Flex__noShrink, noShrink), _cn));
  return /*#__PURE__*/_react["default"].createElement(Container, (0, _extends2["default"])({}, otherProps, {
    style: _objectSpread(_objectSpread(_objectSpread(_objectSpread({}, inlineStyle), marginResult), paddingsResult), style),
    className: classNames,
    "data-test-id": testId
  }), children);
};

exports.Flex = Flex;
