// @contentful/forma-36-react-components/dist/components/Table/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Table", {
  enumerable: true,
  get: function get() {
    return _Table.Table;
  }
});
Object.defineProperty(exports, "TableBody", {
  enumerable: true,
  get: function get() {
    return _TableBody.TableBody;
  }
});
Object.defineProperty(exports, "TableCell", {
  enumerable: true,
  get: function get() {
    return _TableCell.TableCell;
  }
});
Object.defineProperty(exports, "TableHead", {
  enumerable: true,
  get: function get() {
    return _TableHead.TableHead;
  }
});
Object.defineProperty(exports, "TableRow", {
  enumerable: true,
  get: function get() {
    return _TableRow.TableRow;
  }
});
Object.defineProperty(exports, "TableSortingLabel", {
  enumerable: true,
  get: function get() {
    return _TableSortingLabel.TableSortingLabel;
  }
});

var _Table = require("./Table");

var _TableBody = require("./TableBody/TableBody");

var _TableCell = require("./TableCell/TableCell");

var _TableHead = require("./TableHead/TableHead");

var _TableSortingLabel = require("./TableSortingLabel/TableSortingLabel");

var _TableRow = require("./TableRow/TableRow");
