// @contentful/forma-36-react-components/dist/components/Accordion/AccordionItem/AccordionItem.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AccordionItem = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _react = _interopRequireWildcard(require("react"));

var _useId = require("../../../utils/useId");

var _AccordionHeader = require("../AccordionHeader/AccordionHeader");

var _AccordionPanel = require("../AccordionPanel/AccordionPanel");

var styles = {
  "Accordion": "Accordion__Accordion___2vsT6",
  "AccordionItem": "Accordion__AccordionItem___11w9U",
  "Accordion--start": "Accordion__Accordion--start___2hNCQ",
  "AccordionHeader": "Accordion__AccordionHeader___3k1Uh",
  "Accordion--end": "Accordion__Accordion--end___1WaCd",
  "AccordionHeader--expanded": "Accordion__AccordionHeader--expanded___Zmyij",
  "AccordionHeader__icon": "Accordion__AccordionHeader__icon___1pUm4"
};

var AccordionItem = function AccordionItem(_ref) {
  var _ref$title = _ref.title,
      title = _ref$title === void 0 ? 'Accordion Title' : _ref$title,
      _ref$titleElement = _ref.titleElement,
      titleElement = _ref$titleElement === void 0 ? 'h2' : _ref$titleElement,
      _ref$testId = _ref.testId,
      testId = _ref$testId === void 0 ? 'cf-ui-accordion-item' : _ref$testId,
      onExpand = _ref.onExpand,
      onCollapse = _ref.onCollapse,
      children = _ref.children;
  var id = (0, _useId.useId)();

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      isExpanded = _useState2[0],
      setIsExpanded = _useState2[1];

  var onClick = function onClick() {
    if (!isExpanded && onExpand) {
      onExpand();
    }

    if (isExpanded && onCollapse) {
      onCollapse();
    }

    setIsExpanded(!isExpanded);
  };

  return /*#__PURE__*/_react["default"].createElement("li", {
    className: styles.AccordionItem,
    "data-test-id": "".concat(testId, "-").concat(id)
  }, /*#__PURE__*/_react["default"].createElement(_AccordionHeader.AccordionHeader, {
    handleClick: onClick,
    isExpanded: isExpanded,
    element: titleElement,
    ariaId: id
  }, title), /*#__PURE__*/_react["default"].createElement(_AccordionPanel.AccordionPanel, {
    ariaId: id,
    isExpanded: isExpanded
  }, children));
};

exports.AccordionItem = AccordionItem;
