// @contentful/forma-36-react-components/dist/components/Skeleton/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "SkeletonBodyText", {
  enumerable: true,
  get: function get() {
    return _SkeletonBodyText.SkeletonBodyText;
  }
});
Object.defineProperty(exports, "SkeletonContainer", {
  enumerable: true,
  get: function get() {
    return _SkeletonContainer.SkeletonContainer;
  }
});
Object.defineProperty(exports, "SkeletonDisplayText", {
  enumerable: true,
  get: function get() {
    return _SkeletonDisplayText.SkeletonDisplayText;
  }
});
Object.defineProperty(exports, "SkeletonImage", {
  enumerable: true,
  get: function get() {
    return _SkeletonImage.SkeletonImage;
  }
});
Object.defineProperty(exports, "SkeletonRow", {
  enumerable: true,
  get: function get() {
    return _SkeletonRow.SkeletonRow;
  }
});
Object.defineProperty(exports, "SkeletonText", {
  enumerable: true,
  get: function get() {
    return _SkeletonText.SkeletonText;
  }
});

var _SkeletonBodyText = require("./SkeletonBodyText/SkeletonBodyText");

var _SkeletonContainer = require("./SkeletonContainer/SkeletonContainer");

var _SkeletonDisplayText = require("./SkeletonDisplayText/SkeletonDisplayText");

var _SkeletonImage = require("./SkeletonImage/SkeletonImage");

var _SkeletonRow = require("./SkeletonRow/SkeletonRow");

var _SkeletonText = require("./SkeletonText/SkeletonText");
