// @contentful/forma-36-react-components/dist/components/Pill/Pill.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Pill = Pill;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Icon = require("../Icon");

var _TabFocusTrap = require("../TabFocusTrap");

var _excluded = ["label", "onClose", "testId", "onDrag", "className", "dragHandleComponent", "variant"];
var styles = {
  "Pill": "Pill__Pill___2yQFD a11y__focus-outline--default___2hwb1",
  "Pill--deleted": "Pill__Pill--deleted___mwXFF",
  "Pill--idle": "Pill__Pill--idle___3nZ8p",
  "Pill--active": "Pill__Pill--active___3tkL1",
  "Pill__label": "Pill__Pill__label____EsBZ",
  "Pill__label--deleted": "Pill__Pill__label--deleted___1FySm",
  "Pill__label--idle": "Pill__Pill__label--idle___3PU9m",
  "Pill__label--active": "Pill__Pill__label--active___1ZryI",
  "Pill__close-button": "Pill__Pill__close-button___3OlnQ a11y__focus-outline--default___2hwb1",
  "Pill__close-button--deleted": "Pill__Pill__close-button--deleted___s0x2C",
  "Pill__close-button--idle": "Pill__Pill__close-button--idle___OneTW",
  "Pill__close-button--active": "Pill__Pill__close-button--active___3FpQZ",
  "Pill__drag-icon": "Pill__Pill__drag-icon___2aB1g",
  "Pill__icon--deleted": "Pill__Pill__icon--deleted___V8b3O",
  "Pill__icon--idle": "Pill__Pill__icon--idle___1Yi7d",
  "Pill__icon--active": "Pill__Pill__icon--active___11oHX"
};

function Pill(_ref) {
  var label = _ref.label,
      onClose = _ref.onClose,
      _ref$testId = _ref.testId,
      testId = _ref$testId === void 0 ? 'cf-ui-pill' : _ref$testId,
      onDrag = _ref.onDrag,
      className = _ref.className,
      dragHandleComponent = _ref.dragHandleComponent,
      _ref$variant = _ref.variant,
      variant = _ref$variant === void 0 ? 'idle' : _ref$variant,
      otherProps = (0, _objectWithoutProperties2["default"])(_ref, _excluded);
  var classNames = (0, _classnames["default"])(styles.Pill, styles["Pill--".concat(variant)], className);
  return /*#__PURE__*/_react["default"].createElement("div", (0, _extends2["default"])({
    className: classNames,
    "data-test-id": testId
  }, otherProps, {
    onDrag: onDrag
  }), onDrag && (dragHandleComponent ? dragHandleComponent : /*#__PURE__*/_react["default"].createElement("span", {
    className: styles['Pill__drag-icon']
  }, /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: "Drag",
    color: "muted",
    className: styles["Pill__icon--".concat(variant)]
  }))), /*#__PURE__*/_react["default"].createElement("span", {
    "aria-label": label,
    title: label,
    className: (0, _classnames["default"])(styles.Pill__label, styles["Pill__label--".concat(variant)])
  }, label), onClose && variant !== 'deleted' && /*#__PURE__*/_react["default"].createElement("button", {
    type: "button",
    "aria-label": "close",
    onClick: onClose,
    className: (0, _classnames["default"])(styles['Pill__close-button'], styles["Pill__close-button--".concat(variant)])
  }, /*#__PURE__*/_react["default"].createElement(_TabFocusTrap.TabFocusTrap, null, /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: "Close",
    color: "muted",
    className: styles["Pill__icon--".concat(variant)]
  }))));
}
