// @contentful/forma-36-react-components/dist/components/Table/TableCell/TableCell.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.sortingDirections = exports.TableCell = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _TableCellContext = require("./TableCellContext");

var _excluded = ["align", "children", "className", "sorting", "testId"];
var styles = {
  "TableCell--head": "TableCell__TableCell--head___3AHws",
  "TableCell--head__sorting": "TableCell__TableCell--head__sorting___AjdIZ",
  "TableCell": "TableCell__TableCell___ZC2ZI",
  "TableRow--selected": "TableCell__TableRow--selected___15cRw"
};
var sortingDirections = {
  asc: 'asc',
  desc: 'desc'
};
exports.sortingDirections = sortingDirections;

var TableCell = function TableCell(_ref) {
  var _ref$align = _ref.align,
      align = _ref$align === void 0 ? 'left' : _ref$align,
      children = _ref.children,
      className = _ref.className,
      _ref$sorting = _ref.sorting,
      sorting = _ref$sorting === void 0 ? false : _ref$sorting,
      _ref$testId = _ref.testId,
      testId = _ref$testId === void 0 ? 'cf-ui-table-cell' : _ref$testId,
      otherProps = (0, _objectWithoutProperties2["default"])(_ref, _excluded);
  return /*#__PURE__*/_react["default"].createElement(_TableCellContext.TableCellContext.Consumer, null, function (_ref2) {
    var _cn;

    var context = _ref2.name,
        element = _ref2.element,
        offsetTop = _ref2.offsetTop;
    var Element = element;
    return /*#__PURE__*/_react["default"].createElement(Element, (0, _extends2["default"])({
      className: (0, _classnames["default"])(styles['TableCell'], className, (_cn = {}, (0, _defineProperty2["default"])(_cn, styles['TableCell--head'], context === 'head'), (0, _defineProperty2["default"])(_cn, styles['TableCell--head__sorting'], sorting), _cn)),
      style: {
        top: offsetTop
      },
      align: align,
      "data-test-id": testId
    }, otherProps), children);
  });
};

exports.TableCell = TableCell;
